Clarify the use of `:not` with invalid selectors

tlylt opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description
- Improve the explanation to mention that `html` and `body` may be selected unintentionally (it's mentioned in pt 4 but would be good to repeat here)
- Add example showing the effect of using `:not` with invalid selector and how to use `:is` + `:not`

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation
Clarify and add examples.
<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details
nil
<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests
Fixes #24277
<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

tlylt committed:
add example

tlylt committed:
fix wording

estelle commented on files/en-us/web/css/_colon_not/index.md:
````diff
@@ -97,6 +97,41 @@ h2 :not(span.foo) {
+```css
+/* Invalid rule, does nothing */
+:not(.foo, :invalid-pseudo-class) {
+  color: red;
````
```suggestion
  color: red;
  font-style: italic;
```

color blindness: don't want to rely on color alone.

estelle commented on files/en-us/web/css/_colon_not/index.md:
```diff
@@ -97,6 +97,41 @@ h2 :not(span.foo) {
+
+/* Select all <p> elements without the `foo` class */
+p:is(:not(.foo), :not(:invalid-pseudo-class)) {
+  color: green;
```
```suggestion
  color: green;
  border-top: dotted thin currentcolor;
```

estelle reviewed (commented):
Thank you.

I made edits to the examples: we don't want to use colors alone to demonstrate features for accessibility reasons. Also, if the red did work, it would be superceded by green, and everything in the example is a p. Perhaps remove the p from the green example and add it to the red to change which one has greater specificity?

tlylt committed:
improve example

tlylt commented:
Thanks @estelle for the feedback, I have incorporated them.

> I made edits to the examples: we don't want to use colors alone to demonstrate features for accessibility reasons.

Sure, will take note of that in the future.

> Also, if the red did work, it would be superceded by green, and everything in the example is a p. Perhaps remove the p from the green example and add it to the red to change which one has greater specificity?

the p for the green example shows how it does not select the html and body element by accident. I have changed the first paragraph to a div instead so that it won't be affected by the last rule. Hope that's ok.

tlylt committed:
Merge branch 'main' into improve-css-not

estelle reviewed (approved):
Thank you!
